src/pages/projectSettings/tabs/EventLogTab/EventLogDiffs.test.ts

```diff
@@ -0,0 +1,98 @@
+import { getEventDiffLines } from "./EventLogDiffs";
+
+const beforeAddition = {
+  __typename: "ProjectEventSettings",
+  projectRef: {
+    __typename: "Project",
+    identifier: "viewTest",
+    patchTriggerAliases: [],
+  },
+  vars: {
+    __typename: "ProjectVars",
+    vars: { newVariable: "so new" },
+  },
+};
+
+const afterAddition = {
+  __typename: "ProjectEventSettings",
+  projectRef: {
+    __typename: "Project",
+    identifier: "viewTest",
+    patchTriggerAliases: [
+      {
+        __typename: "PatchTriggerAlias",
+        alias: "newAlias",
+        childProjectIdentifier: "evg",
+      },
+    ],
+  },
+  vars: {
+    __typename: "ProjectVars",
+    vars: { newVariable: "so new" },
+  },
+};
+
+const beforeUpdate = {
+  __typename: "ProjectEventSettings",
+  projectRef: {
+    __typename: "Project",
+    identifier: "viewTest",
+    patchTriggerAliases: [
+      {
+        __typename: "PatchTriggerAlias",
+        alias: "newAlias",
+        childProjectIdentifier: "evg",
+      },
+    ],
+  },
+  vars: {
+    __typename: "ProjectVars",
+    vars: { newVariable: "so new" },
+  },
+};
+const afterUpdate = {
+  __typename: "ProjectEventSettings",
+  projectRef: {
+    __typename: "Project",
+    identifier: "viewTest",
+    patchTriggerAliases: [
+      {
+        __typename: "PatchTriggerAlias",
+        alias: "noLongerNewAlias",
+        childProjectIdentifier: "evg",
+      },
+    ],
+  },
+  vars: {
+    __typename: "ProjectVars",
+    vars: { newVariable: "so new" },
+  },
+};
+
+describe("should transform event diffs to key, before and after", () => {
+  it("should transform updates", () => {
+    const diffLines = getEventDiffLines(beforeUpdate, afterUpdate);
+    expect(diffLines).toStrictEqual([
+      {
+        key: "projectRef.patchTriggerAliases[0].alias",
+        before: "newAlias",
+        after: "noLongerNewAlias",
+      },
+    ]);
+  });
+  it("should transform additions", () => {
+    const diffLines = getEventDiffLines(beforeAddition, afterAddition);
+    expect(diffLines).toStrictEqual([
+      {
+        key: "projectRef.patchTriggerAliases[0].alias",
+        before: undefined,
+        after: "newAlias",
+      },
+      {
+        key: "projectRef.patchTriggerAliases[0].childProjectIdentifier",
+        before: undefined,
+        after: "evg",
+      },
```

Comment:
I was contemplating this as well, and I'm not sure that I agree. 
We see both of these examples on the legacy page. 

Here we have it with each on it's own line
<img width="747" alt="image" src="https://user-images.githubusercontent.com/13104717/163838260-5cceb16a-da73-4f85-841b-e1702548ff7d.png">

here we have it all on one line: 
<img width="1122" alt="image" src="https://user-images.githubusercontent.com/13104717/163838322-074f7ac0-ac55-47ec-9376-6954877433f7.png">

In my opinion, listing it all separately makes it easier to visually discern in one quick glance. 

Each event will be separated in its own card. Therefore, it will already be one distinct unit representing the change.

Reply:
I agree -- you only really go into the event log to see how something specific has changed, so having it broken up for that is helpful.

Reply:
True, but wouldn't the latter expand to look more like this than what that first chart looks like?
| Property | Before | After |
|----------|--------|-------|
| proj_ref.owner_name |  | a|
| ... | | |
| pop_ref.patch_trigger_aliases[0].alias |  | newAlias |
| pop_ref.patch_trigger_aliases[0].child_project_id |  | mci |
| pop_ref.patch_trigger_aliases[0].child_project_identifier |  | evg |
| pop_ref.patch_trigger_aliases[0].task_specifiers[0].patch_alias |  | __github |
| pop_ref.patch_trigger_aliases[0].task_specifiers[0].task_regex |  | "" |
| pop_ref.patch_trigger_aliases[0].task_specifiers[0].variant_regex |  | "" |
| pop_ref.patch_trigger_aliases[0].status |  | "" |
| pop_ref.patch_trigger_aliases[0].parent_as_module |  | "" |

To me that comes across as less clear that an entire new patch trigger alias was added. But if you prefer it then that's okay with me!

Reply:
Why don't we first set it up like this, and then maybe iterate on it later if we get user feedback?

Reply:
I tend to like this because for example, it being broken up into lines would make it easier to spot what is wrong with the alias -- for example, the module wasn't added.

Reply:
As for the chart you posted, I think I'll filter out any empty strings when I create the columns to make it clearer.